src/main/java/io/javalin/metrics/JavalinMetrics.java

```diff
@@ -0,0 +1,59 @@
+package io.javalin.metrics;
+
+import io.micrometer.core.instrument.Metrics;
+import io.micrometer.core.instrument.binder.jetty.JettyServerThreadPoolMetrics;
+import io.micrometer.core.instrument.binder.jetty.JettyStatisticsMetrics;
+import io.micrometer.core.instrument.composite.CompositeMeterRegistry;
+import org.eclipse.jetty.server.handler.HandlerWrapper;
+import org.eclipse.jetty.server.handler.StatisticsHandler;
+import org.eclipse.jetty.util.thread.ThreadPool;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Collections;
+import java.util.Optional;
+
+public class JavalinMetrics {
+
+    @Nullable
+    private static JavalinMetrics instance;
+
+    /**
+     * Creates a new {@link JavalinMetrics} instance if needed to prevent multiple and unneeded instantiations.
+     *
+     * @param jettyStatisticsHandler {@link StatisticsHandler} which is already registered via {@link org.eclipse.jetty.server.Server#insertHandler(HandlerWrapper)}
+     * @param jettyThreadPool the {@link ThreadPool} which is used by {@link io.javalin.Javalin}
+     * @return new created or already present instance of {@link JavalinMetrics}
+     */
+    @NotNull
+    public synchronized static JavalinMetrics createInstanceIfNeeded(@NotNull StatisticsHandler jettyStatisticsHandler, @NotNull ThreadPool jettyThreadPool) {
+        if(instance == null) {
+            instance = new JavalinMetrics(jettyStatisticsHandler, jettyThreadPool);
+        }
+
+        return instance;
+    }
+
+    @NotNull
+    public Optional<JavalinMetrics> getInstance() {
```

Comment:
This is currently unused I created it in case someone needs to access the `JavalinMetrics` instance later on. Altough one could argue that we should delete this code because it is not currently used. 
Same applies to the return type of `createInstanceIfNeeded(...)` which is currently not used but might be used later on.

@tipsy What would be your suggestion in that case?

Reply:
My philosophy is to always leave things out until someone requests them. It's easy to add later and release a new version.

Reply:
done